### PR TITLE
[JENKINS-26133] Return exit status or stdout from sh/bat steps

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/durable-task-plugin/pull/27.

- [X] `returnStdout`
- [x] `returnStatus`

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java [hunk 1]
```diff
@@ -58,19 +63,70 @@
...
+        return returnStatus;
+    }
+
+    @DataBoundSetter public void setReturnStatus(boolean returnStatus) {
+        this.returnStatus = returnStatus;
+    }
+
+    public abstract static class DurableTaskStepDescriptor extends AbstractStepDescriptorImpl {
+
+        public static final String defaultEncoding = "UTF-8";
```
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java]: _annoying nit_ constants are usually capitalized.

**Comment by rsandell** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java]: We tend to break that rule for this idiom.

File: src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java [hunk 2]
```diff
@@ -209,8 +275,8 @@ private void check() {
                         LOGGER.log(Level.FINE, "last-minute output in {0} on {1}", new Object[] {remote, node});
                     }
                     t.set(null); // do not interrupt cleanup
-                    if (exitCode == 0) {
-                        getContext().onSuccess(exitCode);
+                    if (returnStatus || exitCode == 0) {
+                        getContext().onSuccess(returnStatus ? exitCode : returnStdout ? new String(controller.getOutput(workspace, launcher), encoding) : null);
                     } else {
                         getContext().onFailure(new AbortException("script returned exit code " + exitCode));
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java]: I guess someone configuring `returnStdout` expects to get the stdout even when the step fails, but in the exception message maybe...

Belated :bug: I guess

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java]: Filed #13.

**Pull request merged**